### PR TITLE
chore: remove device flow

### PR DESCRIPTION
**Pull request by MohammedMokresh**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Update the remove device logic to support cancellation of the process.

### Solutions
update remove device screen and register device screen by adding close button that will clear the session and navigate the user to the welcome screen or the logged in account if any exists

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1247/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5257627)(5257627)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5257627)(5257627)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1247/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5264471)(5264471)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5264471)(5264471)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1247/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5316301)(5316301)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5316301)(5316301)-dev-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceScreen.kt [hunk 1]
```diff
@@ -23,42 +24,81 @@ import androidx.compose.ui.tooling.preview.Preview
...
+    onProceedLoginClicked: () -> Unit
 ) {
+    BackHandler {
+        onBackButtonClicked()
+    }
+    val cancelLoginDialogState = rememberVisibilityState<CancelLoginDialogState>()
+    CancelLoginDialogContent(
+        dialogState = cancelLoginDialogState,
+        onActionButtonClicked = {
+            cancelLoginDialogState.dismiss()
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceScreen.kt]: you call `dismiss()` here and also update the state in ViewModel, do we need this duplication? Why not just in keep it in ViewModel only?

File: app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceScreen.kt [hunk 2]
```diff
@@ -23,42 +24,81 @@ import androidx.compose.ui.tooling.preview.Preview
...
+            onCancelLoginClicked()
+        },
+        onProceedButtonClicked = {
+            cancelLoginDialogState.dismiss()
+            onProceedLoginClicked()
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceScreen.kt:83]: the same here

**Review by borichellow — APPROVED** [review]
good job, just a few small questions/suggestion :)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1247/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5317735)(5317735)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5317735)(5317735)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1247/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5317944)(5317944)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5317944)(5317944)-dev-compat.apk)

**Pull request merged**